framework/sandstone_opts.cpp

```diff
@@ -0,0 +1,846 @@
+        { "output-log", required_argument, nullptr, 'o' },
+        { "quality", required_argument, nullptr, quality_option },
+        { "quick", no_argument, nullptr, quick_run_option },
+        { "quiet", no_argument, nullptr, 'q' },
+        { "retest-on-failure", required_argument, nullptr, retest_on_failure_option },
+        { "rng-state", required_argument, nullptr, 's' },
+        { "schedule-by", required_argument, nullptr, schedule_by_option },
+#ifndef NO_SELF_TESTS
+        { "selftests", no_argument, nullptr, selftest_option },
+#endif
+        { "service", no_argument, nullptr, service_option },
+        { "shorten-runtime", required_argument, nullptr, shortened_runtime_option },
+        { "strict-runtime", no_argument, nullptr, strict_runtime_option },
+        { "syslog", no_argument, nullptr, syslog_runtime_option },
+        { "temperature-threshold", required_argument, nullptr, temperature_threshold_option },
+        { "test-delay", required_argument, nullptr, test_delay_option },
+        { "test-list-file", required_argument, nullptr, test_list_file_option },
+        { "test-range", required_argument, nullptr, test_index_range_option },
+        { "test-list-randomize", no_argument, nullptr, test_list_randomize_option },
+        { "test-time", required_argument, nullptr, 't' },   // repeated below
+        { "force-test-time", no_argument, nullptr, force_test_time_option },
+        { "test-option", required_argument, nullptr, 'O'},
+        { "threads", required_argument, nullptr, 'n' },
+        { "time", required_argument, nullptr, 't' },        // repeated above
+        { "timeout", required_argument, nullptr, timeout_option },
+        { "total-retest-on-failure", required_argument, nullptr, total_retest_on_failure },
+        { "total-time", required_argument, nullptr, 'T' },
+        { "ud-on-failure", no_argument, nullptr, ud_on_failure_option },
+        { "use-builtin-test-list", optional_argument, nullptr, use_builtin_test_list_option },
+        { "vary-frequency", no_argument, nullptr, vary_frequency},
+        { "vary-uncore-frequency", no_argument, nullptr, vary_uncore_frequency},
+        { "verbose", no_argument, nullptr, 'v' },
+        { "version", no_argument, nullptr, version_option },
+        { "weighted-testrun-type", required_argument, nullptr, weighted_testrun_option },
+        { "yaml", optional_argument, nullptr, 'Y' },
+
+#if defined(__SANITIZE_ADDRESS__)
+        { "is-asan-build", no_argument, nullptr, is_asan_option },
+#endif
+#ifndef NDEBUG
+        // debug-mode only options:
+        { "gdb-server", required_argument, nullptr, gdb_server_option },
+        { "is-debug-build", no_argument, nullptr, is_debug_option },
+        { "test-tests", no_argument, nullptr, test_tests_option },
+#endif
+        { nullptr, 0, nullptr, 0 }
+    };
+
+    int opt;
+    int coptind = -1;
+
+    while (!SandstoneConfig::RestrictedCommandLine &&
```

Comment:
This works, but we lose the Dead Code Elimination that this produced in sandstone.cpp: the compiler knows `SandstoneConfig::RestrictedCommandLine` is a constant, so it knew whether the parsing was one was or another, and that some of the variables could only assume certain specific values.

Reply:
Why don't we put it in an `if` instead. And the `if (SandstoneConfig::RestrictedCommandLine)` below will become the `else`. That will work.